wordpress-mobile/WordPress-Android
# Feature/dark mode switch

Fixes #10645 

This PR adds an App Theme preference to App Settings screen.

[![Image from Gyazo](https://i.gyazo.com/235ebc038b8ecc54e40b5fc0c41d4144.png)](https://gyazo.com/235ebc038b8ecc54e40b5fc0c41d4144)

It's a feture branch, so Dark Theme is supposed to look scary, don't mind it :)

To test:

On devices with API <=28:
1. Open App Settings from Me screen.
2. Notice the App Theme preference.
3. The default value should be set to `Set by Battery Saver`.
4. From the preference, dialog select `Dark` and confirm that the app switches to Dark Theme.
5. Select `Light` and confirm that the app switches to Light Theme.
6. Select `Set by Battery Saver` and enable Battery Saver on the device.
7. Confirm that Dark Theme is used when Battery Saver is ON.

On devices with API 29+:
1. Open App Settings from Me screen.
2. Notice the App Theme preference.
3. The default value should be set to `System Default`.
4. From the preference, dialog select `Dark` and confirm that the app switches to Dark Theme.
5. Select `Light` and confirm that the app switches to Light Theme.
6. Select `System Default` and close the app.
7. Enable Dark Theme in Display settings of the device.
8. Launch the app again and confirm that the dark theme is applied.

Note for API 29+: I wasn't able to get the app theme to switch automatically when the Dark Theme is toggled. For now, you need to restart the app for new theme to take effect. I made a separate issue for this at #10748



PR submission checklist:
